packages/DataTable/src/DataTable.js

```diff
@@ -62,14 +62,24 @@ export default function DataTable(props) {
   const columns = ColumnsDefinition.map(Column => Column.props);
 
   let navigationReducers = [];
+  let isControlled = false;
   if (Navigation && Navigation.props) {
     navigationReducers = React.Children.map(Navigation.props.children, child => child.type.reducer).filter(
       chunk => chunk
     );
+    React.Children.forEach(Navigation.props.children, child => {
+      if (child.props.onFilter || child.props.onSort) isControlled = true;
```

Comment:
agree, as we discussed by now, if you set one as controlled, we delegated all the responsibility to the consumer.